app/data/tasks/fetch_record_csv.py

```diff
@@ -0,0 +1,111 @@
+
+
+# Keys which cannot be usefully exported to csv
+DROPPED_KEYS = ['media']
+# static record fields
+RECORD_FIELDS = ['record_id', 'created', 'modified', 'occurred_from',
+                 'occurred_to', 'lat', 'lon', 'location_text',
+                 'city', 'city_district', 'county', 'neighborhood', 'road',
+                 'state', 'weather', 'light']
+
+
+def FIELD_TRANFORMS():
+    def date_iso(d):
+        return d.isoformat()
+
+    FieldTransform = namedtuple('FieldTransform', ['field', 'transform'])
+    return {
+        'record_id': FieldTransform(field='uuid', transform=lambda uuid: str(uuid)),
+        'created': FieldTransform(field='created', transform=date_iso),
+        'modified': FieldTransform(field='modified', transform=date_iso),
+        'occurred_from': FieldTransform(field='occurred_from', transform=date_iso),
+        'occurred_to': FieldTransform(field='occurred_to', transform=date_iso),
+        'lat': FieldTransform(field='geom', transform=lambda geom: geom.y),
+        'lon': FieldTransform(field='geom', transform=lambda geom: geom.x),
+    }
+
+
+@shared_task
+def export_records(occurred_min, occurred_max, record_label=settings.PRIMARY_RECORD_LABEL):
+    def to_utf8(s):
+        """Convert to utf8 encoding and strip special whitespace/commas for csv writing"""
+        if isinstance(s, str):
+            return re.sub(r'[\r\n\t,]', '', s)
+        elif isinstance(s, unicode):
+            return re.sub(r'[\r\n\t,]', '', s).encode('utf-8')
+        elif s is None:
+            return unicode('').encode('utf-8')
+        else:
+            return re.sub(r'[\r\n\t,]', '', unicode(s)).encode('utf-8')
+
+    records = Record.objects.filter(
+        occurred_from__gte=occurred_min,
+        occurred_to__lte=occurred_max,
+        schema__record_type__label=record_label
+    )
+
+    try:
+        record_type = records[0].schema.record_type
+        schema = record_type.get_current_schema()
+    except IndexError:
+        raise Exception('No records in result set')
+
+    jsonschema = schema.schema
+    details = {
+        key: subschema for key, subschema in jsonschema['definitions'].viewitems()
+        if 'details' in subschema and subschema['details'] is True
+    }
+    details_key = details.keys()[0]
+    record_detail_fields = [
+        key for key, val in details[details_key]['properties'].viewitems()
+        if (
+                'options' not in val or
+                'hidden' not in val['options'] or
+                val['options']['hidden'] is False
+        ) and (
+            key not in DROPPED_KEYS
+        )
+    ]
+    csv_columns = RECORD_FIELDS + record_detail_fields
+    transforms = FIELD_TRANFORMS()
+    row_dicts = []
+    for record in records:
+        row = dict()
+        for field in RECORD_FIELDS:
+            if field in transforms:
+                ft = transforms[field]
+                row[to_utf8(field)] = to_utf8(ft.transform(getattr(record, ft.field)))
```

Comment:
Wrapping `field` in `to_utf8()` shouldn't be necessary since they're all hard-coded up top and could potentially introduce unexpected behavior since `to_utf8` also deletes certain characters (which might be a reason to split that functionality out, not sure).